main.go

```diff
@@ -163,14 +163,18 @@ func readFile(fileName string) ([]Entropy, error) {
 }
 
 func entropy(text string) float64 {
-	uniqueCharacters := make(map[rune]struct{}, len(text))
+	uniqueCharacters := make(map[rune]int64, len(text))
 	for _, r := range text {
-		uniqueCharacters[r] = struct{}{}
+		if _, ok := uniqueCharacters[r]; ok {
+			uniqueCharacters[r]++
+		} else {
+			uniqueCharacters[r] = 1
+		}
```

Comment:
You can simply do:

```suggestion
		uniqueCharacters[r]++
```

this works because if uniqueCharacters[r] does not exist, it is read as a `0`, incremented, and then written to `uniqueCharacters[r]`.

Reply:
amazing, i fix that.